x/confidentialtransfers/keeper/msg_server.go

```diff
@@ -358,6 +358,10 @@ func (m msgServer) Transfer(goCtx context.Context, req *types.MsgTransfer) (*typ
 		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "invalid msg")
 	}
 
+	if instruction.FromAddress == instruction.ToAddress {
```

Comment:
Disabled this because the way we write this is
senderState = getAccount(sender)
recipientState = getAccount(recipient)

senderState.Balance += Amount
recipientState.Balance -= Amount

keeper.Store(senderState, sender)
keeper.Store(recipientState, recipient).

If sender and recipient are same address, the updated 'recipientState' overrides the updated 'senderState' for the key 'sender' and funds are not deducted. If we want to enable this we need to write a special case.